src/conventional/sonde_bufr2ioda.py

```diff
@@ -0,0 +1,639 @@
+        elif len(meta_data['latitude']) > 0:
+            target_number = len(meta_data['latitude'])
+    if target_number == 0:
+        print("HOW on earth is target_number zero?  BUFR sucks!")
+        return data, count, start_pos
+
+    count[1] += target_number
+
+    # For any of the MetaData elements that were totally lacking, fill entire vector with missing.
+    empty = []
+    for k, v in metaDataKeyList.items():
+        if not any(meta_data[k]):
+            meta_data[k] = assign_missing_meta(empty, k, target_number, 0)
+        length = len(meta_data[k])
+        if (length == 1):
+            meta_data[k] = np.full(target_number, meta_data[k][0])
+        elif (length < target_number):
+            logging.warning(f"Key called {k} contains only {length} "
+                            f" elements, whereas {target_number} were expected.")
+            meta_data[k] = assign_missing_meta(meta_data[k], k, target_number, length-1)
+        elif (length > target_number):
+            print(f"Key called {k} contains {length} "
+                            f" elements, whereas {target_number} were expected.")
+            meta_data[k] = meta_data[k][:target_number]
+
+    # Sondes are special with a launch time and time displacement.
+    if meta_data['timeDisplacement'].any():
+        meta_data['dateTime'] = specialty_time (bufr, meta_data['timeDisplacement'])
+        meta_data['releaseTime'] = np.full(target_number, meta_data['dateTime'][0])
+    else:
+        count[2] += target_number
+        return data, count, start_pos
+
+    # Sondes also have lat/lon displacement from launch/release location.
+    if (meta_data['latDisplacement'].any() and meta_data['lonDisplacement'].any()):
+        for n, delta_lat in enumerate(meta_data['latDisplacement']):
+            delta_lon = meta_data['lonDisplacement'][n]
+            meta_data['latitude'][n] = meta_data['latitude'][0] + delta_lat
+            meta_data['longitude'][n] = meta_data['longitude'][0] + delta_lon
+    else:
+        count[2] += target_number
+        return data, count, start_pos
+
+    # Force longitude into space of -180 to +180 only. Reset both lat/lon missing if either absent.
+    mask_lat = np.logical_or(meta_data['latitude'] < -90.0, meta_data['latitude'] > 90.0)
+    mask_lon = np.logical_or(meta_data['longitude'] < -180.0, meta_data['longitude'] > 360.0)
+    meta_data['latitude'][mask_lat] = float_missing_value
+    meta_data['longitude'][mask_lon] = float_missing_value
+    meta_data['latitude'][mask_lon] = float_missing_value
+    meta_data['longitude'][mask_lat] = float_missing_value
+    for n, longitude in enumerate(meta_data['longitude']):
+        if (meta_data['longitude'][n] != float_missing_value and meta_data['longitude'][n] > 360):
+            meta_data['longitude'][n] = 360.0 - meta_data['longitude'][n]
+
+    # It is NOT IDEAL, but if missing a lat or lon, fill with prior known value for now.
```

Comment:
Would this be another case to skip this message or put in missing values? It seems a bit risky to silently "repair" bad lat, lon values.

Reply:
Actually this check is enforcing that the longitudes are not permitted to go from zero to 360, but must go per usual convention of -180 to +180 degrees.

However, the point you are making about lines below 523 is correct. I decided to auto-correct missing lat/lon values from the most recently known values due to the fact that the observations are entirely pointless without a lat/lon and the potential error from how far a radiosonde moves in 1 or 2 seconds (of balloon ascent) is pretty small.  In a single example file, I found a few sites that had infrequent spots of missing lat/lon that I felt was worthy of keeping location constant.  This is debatable to fix like this versus using a specialized function to interpolate or use wind info to get a better true estimate.  However, at this time, I would like to move forward as-is with possibility to write a specialized UFO function to fix it there.

Reply:
Thanks for the explanation. I see the point you are making about the lat, lon not changing much between successive readings. There still is a possibility that you could have a long series of bogus lat, lon values resulting in larger errors, but that might be rare, and it seems that the potential function you mention could see that and do a somewhat reasonable job of remedying the situation.

I'm okay with deferring the debate about how/where to correct these values to after this PR.